src/main/java/use_case/zip_search/ZipSearchInteractor.java

```diff
@@ -0,0 +1,121 @@
+// Author: Diego
+package use_case.zip_search;
+
+import entity.University;
+import entity.UniversityFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import use_case.search.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ZipSearchInteractor implements ZipSearchInputBoundary {
+    final ZipSearchUserDataAccessInterface zipSearchDataAccessObject;
+    final ZipSearchOutputBoundary zipSearchPresenter;
+    final UniversityFactory universityFactory;
+    public ZipSearchInteractor(ZipSearchUserDataAccessInterface zipSearchUserDataAccessInterface,
+            ZipSearchOutputBoundary zipSearchPresenter, UniversityFactory universityFactory) {
+        this.zipSearchDataAccessObject = zipSearchUserDataAccessInterface;
+        this.zipSearchPresenter = zipSearchPresenter;
+        this.universityFactory = universityFactory;
+    }
+
+    @Override
+    public void executeSearch(ZipSearchInputData zipSearchInputData) {
```

Comment:
Interactor seems to provide the correct universities given the proper zipSearchInputData given to it.